client/starwhale/api/_impl/dataset/loader.py

```diff
@@ -100,8 +140,23 @@ def _do_iter_row(self) -> t.Generator[TabularDatasetRow, None, None]:
                 if rt is None:
                     break
 
-                for row in self.tabular_dataset.scan(rt[0], rt[1]):
-                    yield row
+                if self.dataset_uri.instance_type == InstanceType.CLOUD:
+                    for btch in self.tabular_dataset.scan_btch(
+                        rt[0], rt[1], self.session_consumption.batch_size  # type: ignore
+                    ):
+                        _links = []
+                        for row in btch:
+                            _links.append(row.data_uri)
+                            _links.extend(_travel_link(row.annotations))
+                        uri_dict = sign_uris([lk.uri for lk in _links])
+                        for lk in _links:
+                            lk.signed_uri = uri_dict.get(lk.uri) or ""  # type: ignore
+
```

Comment:
```python
lk.signed_uri = uri_dict.get(lk.uri) or "" # type: ignore
```
->
```python
lk.signed_uri = uri_dict.get(lk.uri, "")
```
then we can remove `#type: ignore`.

Reply:
done